### PR TITLE
[onert/draft] Optimizer

### PR DESCRIPTION
**Pull request by ragmani**
ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: runtime/onert/backend/train/Tensor.cc [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+// TODO Optimize this method
+void TrainableTensor::applyGradient(const IPortableTensor &grad_tensor, double lr)
+{
+  assert(data_type() == grad_tensor.data_type());
+
+  const auto &shape = get_info().shape();
+  const auto &grad_shape = grad_tensor.get_info().shape();
+
+  // TODO Support for different shapes
```
Comment (ragmani) [inline runtime/onert/backend/train/Tensor.cc:55]: We have to consider applying gradient for each operation.

File: runtime/onert/backend/train/Tensor.h [hunk 2]
```diff
@@ -26,8 +27,42 @@ namespace backend
...
 {
 
-using Tensor = basic::Tensor;
-using ExternalTensor = basic::ExternalTensor;
+// NOTE This class can be replaced with basic::Tensor if this backend supports dynamic tensors.
+class Tensor : public basic::Tensor
+{
+public:
+  Tensor() = delete;
+  ~Tensor();
```
Comment (jyoungyun) [inline runtime/onert/backend/train/Tensor.h:35]: ~Tensor() = default;

File: runtime/onert/backend/train/Tensor.h [hunk 3]
```diff
@@ -26,8 +27,42 @@ namespace backend
...
+
+public:
+  bool applyShape(const ir::Shape &) override { return false; }
+};
+
+class TrainableTensor : public Tensor, public train::ITrainableTensor
+{
+public:
+  TrainableTensor() = delete;
+  ~TrainableTensor();
```
Comment (jyoungyun) [inline runtime/onert/backend/train/Tensor.h:52]:
```suggestion
  ~TrainableTensor() = default;
```

File: runtime/onert/api/src/nnfw_api_internal.cc [hunk 4]
```diff
@@ -195,11 +195,31 @@ std::unique_ptr<onert::ir::Model> loadModel(const std::string filename,
...
 nnfw_session::nnfw_session()
-  : _nnpkg{nullptr}, _coptions{}, _compiler_artifact{nullptr}, _execution{nullptr},
-    _kernel_registry{nullptr}
+  : _nnpkg{nullptr}, _coptions{}, _compiler_artifact{nullptr}, _execution{nullptr}, _kernel_registry
+{
+  nullptr
+}
+#ifdef ONERT_TRAIN
+, _training_step { 0 }
+#endif // ONERT_TRAIN
```
Comment (ragmani) [inline runtime/onert/api/src/nnfw_api_internal.cc:222]: The format checker makes this weird :cry:

**Comment by ragmani** [comment]
Done.

**Pull request closed without merging**